src/prefect/cli/describe.py

```diff
@@ -0,0 +1,225 @@
+
+    \b
+    Usage:
+        $ prefect describe [OBJECT]
+
+    \b
+    Arguments:
+        flow-runs   Describe flow runs
+        flows       Describe flows
+        tasks       Describe tasks
+
+    \b
+    Examples:
+        $ prefect describe flows --name My-Flow --version 2
+        {
+            "name": "My-Flow",
+            "version": 2,
+            "project": {
+                "name": "Test-Project"
+            },
+            "created": "2019-05-08T23:04:58.984132+00:00",
+            "description": null,
+            "parameters": [],
+            "archived": false,
+            "storage": {
+                "type": "Docker",
+                "flows": {
+                    "My-Flow": "/root/.prefect/My-Flow.prefect"
+                },
+                "image_tag": "944444e8-8862-4d04-9e36-b81ab15dcaf6",
+                "image_name": "z4f0bb62-8cc1-49d9-bda3-6rf53b865ea5",
+                "__version__": "0.5.3",
+                "registry_url": "myregistry.io/flows/"
+            },
+            "environment": {
+                "type": "CloudEnvironment",
+                "__version__": "0.5.3"
+            }
+        }
+    """
+    pass
+
+
+@describe.command(hidden=True)
+@click.option("--name", "-n", required=True, help="A flow name to query.", hidden=True)
+@click.option("--version", "-v", type=int, help="A flow version to query.", hidden=True)
+@click.option("--project", "-p", help="The name of a project to query.", hidden=True)
+def flows(name, version, project):
+    """
+    Describe a Prefect flow.
+
+    \b
+    Options:
+        --name, -n      TEXT    A flow name to query                [required]
+        --version, -v   INTEGER A flow version to query
+        --project, -p   TEXT    The name of a project to query
+    """
+    query = {
+        "query": {
+            with_args(
+                "flow",
+                {
+                    "where": {
+                        "_and": {
+                            "name": {"_eq": name},
+                            "version": {"_eq": version},
+                            "project": {"name": {"_eq": project}},
+                        }
+                    },
+                    "order_by": {
+                        "name": EnumValue("asc"),
+                        "version": EnumValue("desc"),
+                    },
+                    "distinct_on": EnumValue("name"),
+                },
+            ): {
+                "name": True,
+                "version": True,
+                "project": {"name": True},
+                "created": True,
+                "description": True,
+                "parameters": True,
+                "archived": True,
+                "storage": True,
+                "environment": True,
+            }
+        }
+    }
+
+    result = Client().graphql(query)
+
+    flow_data = result.data.flow
+
+    if flow_data:
+        click.echo(flow_data[0])
```

Comment:
What happens if the query returns multiple flows (for example, if I don't provide a version or project and my flow has multiple versions)?  Should you include an `order_by` clause in the graphql query so it defaults to returning the lastest version?

Reply:
It goes by flow run name, if you think that we may have conflicting randomly generated flow run names I can add an order by. Another improvement I want to add in a later PR is ID specification for all commands

Reply:
This command I'm referring to is for `describe flows` not flow runs

Reply:
There is an `order_by` in the query
```
                    "order_by": {
                        "name": EnumValue("asc"),
                        "version": EnumValue("desc"),
                    },
```